lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -123,41 +126,78 @@ public function convertException($message, DriverException $exception)
 
     /**
      * {@inheritdoc}
+     *
+     * @return AbstractPlatform|MariaDb102Platform|MySQL57Platform|MySqlPlatform
+     * @throws DBALException
      */
-    public function createDatabasePlatformForVersion($version)
+    public function createDatabasePlatformForVersion($version) : AbstractPlatform
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
+        if (false !== stripos($version, 'mariadb')) {
+            $versionNumber = $this->getMariaDbMysqlVersionNumber($version);
+            if (version_compare($versionNumber, '10.2.7', '>=')) {
+                return new MariaDb102Platform();
+            }
+        } else {
+            $versionNumber = $this->getOracleMysqlVersionNumber($version);
+            if (version_compare($versionNumber, '5.7.9', '>=')) {
+                return new MySQL57Platform();
+            }
+        }
+
+        return $this->getDatabasePlatform();
+    }
+
+    /**
+     * Get a normalized 'version number' from the server string
+     * returned by Oracle MySQL servers.
+     *
+     * @param string $versionString Version string returned by the driver, i.e. '5.7.10'
+     * @throws DBALException
+     */
+    private function getOracleMysqlVersionNumber(string $versionString) : string
+    {
+        if (!preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $versionString, $versionParts)) {
```

Comment:
We can extract the regex into a private const to make things more readable.

Reply:
Great, but very coupled to the exception message (excepted format: <major>.<minor>.<patch>. I'll do anyway. Thanks